src/util/common.rs

```diff
@@ -60,54 +74,36 @@ pub(crate) fn saturate_day_in_month(
 ///
 /// This correctly returns `29` when the year is a leap year and the month is
 /// February.
+#[inline]
 pub(crate) fn days_in_month(year: Year, month: Month) -> Day {
-    if month == 2 && is_leap_year(year) {
-        Day::N::<29>()
-    } else {
-        days_in_month_common_year(month)
-    }
+    Day::new_unchecked(days_in_month_unranged_unchecked(
+        year.get(),
+        month.get() as u8,
+    ))
```

Comment:
You've gotta stay in ranged integer land here. As soon as you start creating new ranged integers like this, the constructor can't tell what the min/max values are allowed to be, and so it assumes the worst case.

There's a reason why there is duplicated code in this module. You can't use unranged code to implemented arithmetic for ranged integers, and you can't use ranged integers inside `const` functions.

Reply:
Fixed in https://github.com/BurntSushi/jiff/pull/23/commits/bfefca72b020141c7478a2f877a0d9eff1857410.

FYI just adding `year.get()` in `days_in_month` panics with
```
thread 'civil::iso_week_date::tests::prop_next_day_is_greater' panicked at src/util/rangeint.rs:2173:1:
ri16 max 10000 is not in range -9999..=9999
```